packages/storage/amplify_storage_s3/example/integration_test/download_data_test.dart

```diff
@@ -0,0 +1,261 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:convert';
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_storage_s3/amplify_storage_s3.dart';
+import 'package:amplify_storage_s3_example/amplifyconfiguration.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+import 'utils/configure.dart';
+import 'utils/sign_in_new_user.dart';
+
+void main() {
+  late StorageUploadDataResult bytesUploadResult;
+  late String bytesPath;
+  late List<int> bytesData;
+
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('downloadData()', () {
+    setUpAll(() async {
+      await configure(amplifyEnvironments['main']!);
+
+      bytesPath = 'public/download-data-${uuid()}';
+      bytesData = 'test data'.codeUnits;
+      addTearDown(
+        () => Amplify.Storage.remove(path: StoragePath.fromString(bytesPath))
+            .result,
+      );
+
+      bytesUploadResult = await Amplify.Storage.uploadData(
+        path: StoragePath.fromString(bytesPath),
+        data: HttpPayload.bytes(bytesData),
+      ).result;
+    });
+
+    group('for data type', () {
+      test('bytes', () async {
+        expect(bytesUploadResult.uploadedItem.path, bytesPath);
+
+        final downloadResult = await Amplify.Storage.downloadData(
+          path: StoragePath.fromString(bytesPath),
+        ).result;
+
+        expect(downloadResult.bytes, bytesData);
```

Comment:
todo: Add an expect for `downloadedItem.path`. Make this change for all tests in the PR. We want to make sure results have the correct path on the storage item.

```dart
expect(downloadResult.downloadedItem.path, bytesPath);
```